compiler/src/dotty/tools/dotc/parsing/Scanners.scala

```diff
@@ -197,7 +197,6 @@ object Scanners {
     val noindentSyntax =
       ctx.settings.noindent.value
       || ctx.settings.oldSyntax.value
-      || migrateTo3
```

Comment:
I think having `-source 3.0-migrate` imply `-noindent` is intentional: it should allow existing weirdly formatted scala 2 code to compile /cc @odersky.

Reply:
The issue is that if a user run compiler with `-rewrite`, then it's impossible to compile the code again.

It's discovered when we actually compile the rewritten code in the following test:

```Scala
compileFile("tests/rewrites/rewrites.scala", scala2CompatMode.and("-rewrite", "-indent"))
```

Reply:
In that example I think `-indent` needs to take precedence other `migrateTo3` and force indentation to be on.